### PR TITLE
Handle duplicate route error gracefully

### LINKED ISSUE #519: Handle duplicate route API error
#### Overview and problem statement

When pushing a worker with a different name to a route that already exists, the Cloudflare Workers API returns an error. Wrangler should handle this error gracefully, and not print the raw API response.

#### Basic example

To reproduce, create a worker named `worker`, change your route to `myzone.com/test` run `wrangler publish`. Then change the name in your toml to `worker-test` and run `wrangler publish` again. You should see this: 

```sh
Error: ⚠️  There was an error creating your route.
 Status Code: 409 Conflict
 Msg: {
  "result": null,
  "success": false,
  "errors": [
    {
      "code": 10020,
      "message": "workers.api.error.duplicate_route"
    }
  ],
  "messages": []
}
```

It'd be nicer if we had a map of status codes/API messages to more user friendly messages.

### PR DESCRIPTION
**Pull request by pradovic**
Should fix #519. I took a really easy & straightforward approach here, as this is my first PR. Let me know if you guys would like it to be solved differently, and I will be more then happy to update the PR or create a new one.

🌷 Maybe it would be useful to add tests for `route.rs`. Would be happy to do it.

### CODE REVIEW

**Comment by pradovic** [comment]
hey @ashleygwilliams, thanks for the super-fast and super-kind response!
I have started writing the unit tests, but I don't see a way to implement proper unit tests without partial refactor of the original code, as it is calling public methods directly, and not really working with traits.

Specifically, it directly calls `http::auth_client` which can not be easily mocked.
Would it be OK to merge this PR without the tests, as I will create a separate PR with the minor refactor + the tests for this one?

Much appreciated!

**Comment by ashleygwilliams** [comment]
@pradovic sounds liek a plan to me! do you want to file an issue for the plan you have! i think my team would be interested in discussing it with you :) ... in the meantime, we should be able to get this in for the next release, 1.6.0- which is currently aiming to be released on nov 8! thanks again for contributing!! <3

**Comment by pradovic** [comment]
@ashleygwilliams Sounds good to me! I will open an issue very soon! Thanks a lot!

File: src/commands/publish/route.rs [hunk 1]
```diff
@@ -98,12 +98,22 @@ fn create(user: &GlobalUser, target: &Target, route: &Route) -> Result<(), failu
         .send()?;
 
     if !res.status().is_success() {
-        let msg = format!(
-            "{} There was an error creating your route.\n Status Code: {}\n Msg: {}",
+        let msg;
+        if res.status().as_u16() == 10020 {
+            msg = format!(
+            "{} There was an error creating your route. Worker with a different name was found on {} route.\n",
```
[the remark this answers is not in the document]
Comment (pradovic) [reply src/commands/publish/route.rs]: Hey @EverlastingBugstopper, thanks for the feedback, I was wondering how to phrase it better! I have updated the PR with suggested change.

**Comment by ashleymichal** [comment]
@pradovic Thanks for the PR! We're definitely looking at improving the testability of a lot of our codebase. One of my big initiatives for this quarter is to shift all of our API calls into cloudflare-rs, which provides a lovely mock api client struct for exactly these kinds of unit tests. i'll probably link to that initiative from this issue: https://github.com/cloudflare/wrangler/issues/63, so keep an eye out if you'd like to help with that!

**Comment by pradovic** [comment]
@ashleymichal Thanks a lot for the info, that sounds awesome! I will keep an eye on that one!

**Pull request merged**